Mage/Command/Factory.php

```diff
@@ -43,14 +43,17 @@ public static function get($commandName, Config $config)
             // try a custom command
             $className = 'Command\\' . $commandName;
 
+            // TODO use a custom exception
             if (!class_exists($className)) {
                 throw new Exception('Command "' . $commandName . '" not found.');
             }
         }
 
         /** @var AbstractCommand $instance */
+        // TODO dependencies like $config should be injected into constructor
```

Comment:
Not exactly. At first, we have no DI mechanism in Magallanes, and as @andres-montanez said in one of issue, we shouldn't use external dependencies as long as Magallanes lives also as a Pear package.
But injecting things in many DI mechanisms can be done by constructor or via proper methods. Sometimes it's even necessary to inject them like that.